src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -744,14 +744,31 @@ internal static bool IsPackageReferenceProject(Project project)
                     if (matchingPackages.Any())
                     {
                         var topLevelPackage = matchingPackages.Single();
-                        InstalledPackageReference installedPackage;
+                        InstalledPackageReference installedPackage = default;
 
                         //If the package is not auto-referenced, get the version from the project file. Otherwise fall back on the assets file
                         if (!topLevelPackage.AutoReferenced)
                         {
                             try
                             { // In case proj and assets file are not in sync and some refs were deleted
-                                installedPackage = projPackages.Where(p => p.Name.Equals(topLevelPackage.Name, StringComparison.Ordinal)).First();
+                                if (assetsFile.PackageSpec.RestoreMetadata.CentralPackageVersionsEnabled)
+                                {
+                                    foreach (KeyValuePair<string, CentralPackageVersion> packageCentralVersion in tfmInformation.CentralPackageVersions)
+                                    {
+                                        if (packageCentralVersion.Key.Equals(topLevelPackage.Name, StringComparison.Ordinal))
```

Comment:
Does CPM require package ID's to be case sensitive?

Suppose in Directory.packages.props you have one casing (e.g.:  NuGet.Common), but in your project-level override you have a different casing (e.g.:  nuget.common).  Would CPM support this?  It looks like this line of code does not.